source/common/lua/lua.cc

```diff
@@ -71,7 +71,7 @@ uint64_t ThreadLocalState::registerGlobal(const std::string& global) {
     if (lua_isfunction(tls.state_.get(), -1)) {
       tls.global_slots_.push_back(luaL_ref(tls.state_.get(), LUA_REGISTRYINDEX));
     } else {
-      ENVOY_LOG(warn, "no '{}' global found in script", global);
+      ENVOY_LOG(info, "definition for '{}' not found in script", global);
```

Comment:
@mattklein123  changed this. see if text makes sense. While I am here, want to understand couple of things
1. Why is this log line coming 9 times if we enable this script for one connection manager?
2. Another info logline  "registering new type: N5Envoy4Http6Filter3Lua19StreamHandleWrapperE" coming multiple times.

Reply:
> Why is this log line coming 9 times if we enable this script for one connection manager? 

Because the symbol is looked up both at initial load as well as per-worker (you probably have 8 workers). You could potentially rework how the logging works if you are inclined, but not sure it's worth it.

> Another info logline "registering new type: N5Envoy4Http6Filter3Lua19StreamHandleWrapperE" coming multiple times.

I would reduce that message to debug level.

Reply:
@mattklein123 Ok. Thanks,
1. I was trying to see if there is an easy way to identify if the current thread is the main thread and log only if it is main thread. But looks there is no easy way unless adding additional method to expose `main_thread_id_` from `Instance` and expose `parent_` in `Slot`. Not sure if would be useful for other use cases. If yes, it would be good to expose them otherwise for this use case it may be an overkill. let me know if there is any other easy way to identify main thread.
2. I have changed it to debug.

Reply:
I think what you did is fine for now. I will take a look and see if I can make the logging a bit better when I look at the headers::add issue.

Reply:
Ok. Thanks